### PR TITLE
Add comments and explanation to joint_diagnostic

### PR DESCRIPTION
**Pull request by davetcoleman**
Added explanation of what the objective is suppose to do, since its not obvious you have to have almost identical waypoint names. Inline documentation!

### CODE REVIEW

**Comment by davetcoleman** [comment]
I think this should also be copied to our Mujoco config

File: src/picknik_ur_mock_hw_config/objectives/joint_diagnostic.xml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <root BTCPP_format="4" main_tree_to_execute="Joint Diagnostic">
   <!-- ////////// -->
-  <BehaviorTree ID="Joint Diagnostic" _description="Move to a known pose and cycle through the min and max limits of a joint repeatedly" _favorite="true">
+  <BehaviorTree ID="Joint Diagnostic" _description="Example of cycliing through the min and max limits of a single joint, for hardware testing" _favorite="true">
```
Comment (bgill92) [inline src/picknik_ur_mock_hw_config/objectives/joint_diagnostic.xml:4]:
```suggestion
  <BehaviorTree ID="Joint Diagnostic" _description="Example of cycling through the min and max limits of a single joint, for hardware testing" _favorite="true">
```

File: src/picknik_ur_mock_hw_config/objectives/joint_diagnostic.xml [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
     <Control ID="Sequence" name="TopLevelSequence">
+      <!--Setup by moving home-->
       <SubTree ID="Move to Waypoint" waypoint_name="Home" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" planner_interface="moveit_default"/>
       <Decorator ID="KeepRunningUntilFailure">
         <Control ID="Sequence">
+          <!-- If you use two waypoints that have idential joint values, except for one joint, you can use this Objective as a way to test a single joint repeatidly. This is useful for things like hardware diagnostics and hardware bringup.-->
```
Comment (bgill92) [inline src/picknik_ur_mock_hw_config/objectives/joint_diagnostic.xml:10]:
```suggestion
          <!-- If you use two waypoints that have identical joint values, except for one joint, you can use this Objective as a way to test a single joint repeatedly. This is useful for things like hardware diagnostics and hardware bringup.-->
```

**Review by bgill92 — CHANGES_REQUESTED** [review]
Looks good, other than spelling typos.

**Comment by bgill92** [comment]
@davetcoleman I can add this `Joint Diagnostic` Objective to the MuJoCo config as well.

**Comment by davetcoleman** [comment]
I just turned on a spell check extension to VSCode :-) 

Thanks for the review! I assume this is already in Mujoco config, just out of date?

**Comment by bgill92** [comment]
Yup, the `Joint Diagnostic` Objective is in the MuJoCo config, just not updated.

**Comment by davetcoleman** [comment]
Seems like the CI is broken unrelated to this PR?

```
 picknik_ur_studio_integration_testing.TestGTestProcessPostShutdown test_gtest_pass
  <<< failure message
    Traceback (most recent call last):
      File "/__w/moveit_studio_ur_ws/moveit_studio_ur_ws/src/picknik_ur_studio_integration_testing/test/launch/test_get_planning_groups.test.py", line 62, in test_gtest_pass
        launch_testing.asserts.assertExitCodes(
      File "/opt/ros/humble/lib/python3.10/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
        assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
    AssertionError: Proc test_get_planning_groups-21 exited with code 1
  >>>

Summary: 136 tests, 0 errors, 8 failures, 14 skipped
```

**Review by MikeWrock — APPROVED** [review]

**Comment by davetcoleman** [comment]
@bgill92 is this CI issue known? Should we just merge and ignore it?

**Pull request merged**